### ISSUE TITLE
Add support for Last-Modified header

### ISSUE THREAD

**Issue by dmitrizagidulin** [solid/solid#10]
If a resource is readable, the `Last-Modified` header should be returned.

(Originally opened by @csarven)
(Moved from https://github.com/linkeddata/gold/issues/64)

From RFC 7232:

> An origin server SHOULD send Last-Modified for any selected
   representation for which a last modification date can be reasonably
   and consistently determined, since its use in conditional requests
   and evaluating cache freshness ([RFC7234]) results in a substantial
   reduction of HTTP traffic on the Internet and can be a significant
   factor in improving service scalability and reliability.
https://tools.ietf.org/html/rfc7232#section-2.2.1

**Comment by dmitrizagidulin** [comment]
@csarven - do you still consider this an important feature request; should we move this issue to /specification ?

**Comment by csarven** [comment]
I think it is important for an implementation but the Solid spec most likely won't change anything from RFC 7232 - currently isn't. If there is any interest in it being something other than a SHOULD for Solid, then sure we can have an issue and explore.

**Comment by kjetilk** [comment]
Yeah, anything that can enable caching is important for Solid, IMHO, so yeah, important, but also possibly a SHOULD with us doing it in our server code to set an example. :-) I believe it is more important to think about it in the server architecture than in the spec.
